ascii/ascii.go

```diff
@@ -74,8 +84,11 @@ func NewASCII(dat string) (*ASCII, error) {
 				break
 			}
 			if v == ',' {
-				chompState, bg = 2, -1
-				continue
+				if len(dat) > i+1 && dat[i+1] >= '0' && dat[i+1] <= '9' {
```

Comment:
peaking ahead like this breaks the fsm formulation; I think the handling should be in `case 2` as like ` if v >= ...  {} else if bgs == 0 { chompState = 0; break }`